src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,223 @@
+        btTab.add("Current Battery Voltage", 0).withSize(2, 2).withPosition(0, 0).getEntry();
+    sbSimVoltage = btTab.add("Simulation Voltage", 0).withSize(2, 2).withPosition(2, 0).getEntry();
+    sbTimer = btTab.add("Timer", 0).withSize(2, 2).withPosition(0, 2).getEntry();
+    sbTimerHighCurrent =
+        btTab.add("High Current Timer", 0).withSize(2, 2).withPosition(0, 4).getEntry();
+    sbTimerChange = btTab.add("Change Timer", 0).withSize(2, 2).withPosition(4, 0).getEntry();
+    timer.start();
+    // currentTimer.start();
+    DriverStationSim.setSendError(true);
+    // btTab.addBoolean("Change the battery: ", valueSupplier)
+  }
+
+  public void periodic() {
+    sbVoltage.setDouble(getVoltage());
+    // sbInCurrent.setDouble(getInputCurrent());
+    sbSimVoltage.setNumber(RoboRioSim.getVInVoltage()); // Just temporary, replace with line above
+    sbTimer.setDouble(timer.get());
+    sbTimerHighCurrent.setDouble(currentTimer.get());
+    sbTimerChange.setBoolean(checkTimer()); // Replace checkVoltage() with checkTimer() if necessary
+    checkCurrent();
+    checkVoltage();
+    // checkMinute();
+  }
+
+  public double getVoltage() {
+    if (Robot.isSimulation()) {
+      return RoboRioSim.getVInVoltage();
+    } else {
+      return RobotController.getBatteryVoltage();
+    }
+  }
+
+  public double getInputCurrent() {
+    return RobotController.getInputCurrent();
+  }
+
+  public void checkCurrent() {
+    if (getInputCurrent() < Constants.maxBatteryCurrent) {
+      stopHCTimer();
+      // currentTimer
+      //    .stop(); // Was current timer, but I got a warning about it needing to be static so I
+      // changed it to this. Will this cause conflicts?
+    } else {
+      startHCTimer();
+      // currentTimer.start();
+    }
+  }
+
+  public boolean checkTimer() {
+    if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentRed)) {
+      DriverStation.reportWarning("Change the Battery Now!", false);
+      // playAudio(Constants.audiofilepath);
+      return true;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralRed)) {
+      DriverStation.reportWarning("Change the Battery Now!", false);
+      // playAudio(Constants.audiofilepath);
+      return true;
+    } else if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon!", false);
+      // playAudio(Constants.audiofilepath);
+      return false;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon!", false);
+      // playAudio(Constants.audiofilepath);
+      return false;
+    }
+    return false;
+  }
+
+  /*
+  public void checkMinute() {
+    double check = (getTimer() % 60.0);
+    if (check > 0 & check < 0.025 & getTimer() > 1.0) {
+      minutesPassed += 1;
+    } else {
+    }
+  }
+  */
+
+  public int getMinute() {
+    return (int) Math.floor(timer.get() / 60.0);
+  }
```

Comment:
Clean up unused methods and commented out code

Reply:
done